pkg/compiler/transformers/k8s/node.go

```diff
@@ -221,6 +221,18 @@ func buildTasks(tasks []*core.CompiledTask, errs errors.CompileErrors) map[commo
 				errs.Collect(errors.NewValueCollisionError(taskID, "Id", taskID))
 			}
 
+			// We don't want Annotation data available at task runtime for performance.
+			if flyteTask.Template != nil &&
+				flyteTask.Template.Interface != nil &&
+				flyteTask.Template.Interface.Inputs != nil &&
+				flyteTask.Template.Interface.Inputs.Variables != nil {
+				for _, v := range flyteTask.Template.Interface.Inputs.Variables {
```

Comment:
We should probably do the same with outputs for consistency/completeness sake....

Reply:
addressed with new commit